js/geo/lng_lat.js

```diff
@@ -82,8 +82,8 @@ LngLat.prototype.toString = function () {
 LngLat.convert = function (input) {
     if (input instanceof LngLat) {
         return input;
-    } else if (input && input.lng && input.lat) {
-        return input;
+    } else if (typeof input !== 'undefined' && input.hasOwnProperty('lng') && input.hasOwnProperty('lat')) {
```

Comment:
`{}` is truthy, so no need for the first typeof. Also, I think you can safely use `'lng' in input` instead of `hasOwnProperty` because it doesn't matter in this case if it's own or inherited from a prototype.